### PR TITLE
{Compute} `az sig image-version create`: Change `--virtual-machine` parameter map to new VM backend field

### PR DESCRIPTION
**Pull request by Jing-song**
**Related command**

`az sig image-version create`

**Description**

close https://github.com/Azure/azure-cli/issues/28488

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Comment by zhoxing-ms** [comment]
@ReaNAiveD Could you please help review this PR?

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -4782,6 +4782,8 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         source = os_disk_image = data_disk_images = None
         if managed_image is not None:
             source = GalleryArtifactVersionFullSource(id=managed_image)
+        if virtual_machine is not None:
+            source = GalleryArtifactVersionFullSource(virtual_machine_id=virtual_machine)
```
Comment (ReaNAiveD) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4786]: Accord to the `L2425-L2434` in file `vm/_validators.py`, the `virtual_machine` value is also assigned to `managed_image`. Consequently, the line `source = GalleryArtifactVersionFullSource(id=managed_image)` will be executed even when a `--virtual-machine` argument is provided. This overlap could introduce potential risks.

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 2]
```diff
@@ -4782,6 +4782,8 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
         source = os_disk_image = data_disk_images = None
         if managed_image is not None:
             source = GalleryArtifactVersionFullSource(id=managed_image)
+        if virtual_machine is not None:
```
Comment (ReaNAiveD) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:4785]: The issue mentioned the min API version is 2023-07-03. Should we wrap the code with min_api check?
image

Comment (Jing-song) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py]: Yes,added.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**